velox/functions/lib/DateTimeFormatter.h

```diff
@@ -0,0 +1,403 @@
+  /// being 1 the formatted result will be 1, with min digit being 4 the
+  /// formatted result will be 0001
+  DateTimeFormatterBuilder& appendClockHourOfDay(size_t minDigits);
+
+  /// Appends minute of hour to formatter builder, e.g: 59
+  ///
+  /// \param minDigits describes the minimum number of digits this format is
+  /// required to represent minute of hour. The format by default is going
+  /// use as few digits as possible greater than or equal to minDigits to
+  /// represent minute of hour. e.g. 1999-01-01 01:59:00, with min digit
+  /// being 1 the formatted result will be 59, with min digit being 4 the
+  /// formatted result will be 0059
+  DateTimeFormatterBuilder& appendMinuteOfHour(size_t minDigits);
+
+  /// Appends second of minute to formatter builder, e.g: 59
+  ///
+  /// \param minDigits describes the minimum number of digits this format is
+  /// required to represent second of minute. The format by default is going
+  /// use as few digits as possible greater than or equal to minDigits to
+  /// represent second of minute. e.g. 1999-01-01 01:59:00, with min digit
+  /// being 1 the formatted result will be 59, with min digit being 4 the
+  /// formatted result will be 0059
+  DateTimeFormatterBuilder& appendSecondOfMinute(size_t minDigits);
+
+  /// Appends fraction of second to formatter builder, e.g: 998
+  ///
+  /// \param digits describes the number of digits this format uses to represent
+  /// fraction of second. e.g. 1999-01-01 01:59:00.987, with digit being 1 the
+  /// formatted result will be 9, with digit being 4 the formatted result will
+  /// be 9870, with digit being 6 the formatted result will be 987000
+  DateTimeFormatterBuilder& appendFractionOfSecond(size_t digits);
+
+  /// Appends time zone to formatter builder, e.g: 'Pacific Standard Time' or
+  /// 'PST'
+  ///
+  /// \param minDigits describes the minimum number of digits this format is
+  /// required to represent time zone. The format by default is going
+  /// use as few digits as possible greater than or equal to minDigits to
+  /// represent time zone.
+  DateTimeFormatterBuilder& appendTimeZone(size_t minDigits);
+
+  /// Appends time zone offset id to formatter builder, e.g: '-0800' or '-08:00'
+  /// or 'America/Los_Angeles'
+  ///
+  /// \param minDigits describes the minimum number of digits this format is
+  /// required to represent time zone offset id. The format by default is going
+  /// use as few digits as possible greater than or equal to minDigits to
+  /// represent time zone offset id.
+  DateTimeFormatterBuilder& appendTimeZoneOffsetId(size_t minDigits);
+
+  DateTimeFormatterBuilder& appendLiteral(const std::string_view& literal);
+
+  DateTimeFormatterBuilder& appendLiteral(
+      const char* literalStart,
+      size_t literalSize);
+
+  std::shared_ptr<DateTimeFormatter> build();
+
+ private:
+  std::unique_ptr<char[]> literalBuf_;
+  size_t bufEnd_{0};
```

Comment:
Isn't that bufSize_, actually?
And the actual size of literalBuf_ is capacity?
Why the way, where do we store literalBuf_'s capacity?